wpilibj/src/main/java/edu/wpi/first/wpilibj/RobotBase.java

```diff
@@ -158,6 +160,52 @@ protected RobotBase() {
       System.err.println("timed out while waiting for NT server to start");
     }
 
+    Consumer<NetworkTableEvent> dashboardConsumer =
+        new Consumer<NetworkTableEvent>() {
+          private boolean m_detected;
+
+          @Override
+          public void accept(NetworkTableEvent t) {
+            if (t.connInfo.remote_id.startsWith("glass")) {
+              HAL.report(tResourceType.kResourceType_Dashboard, tInstances.kDashboard_Glass);
+              m_detected = true;
+            } else if (t.connInfo.remote_id.startsWith("SmartDashboard")) {
+              HAL.report(
+                  tResourceType.kResourceType_Dashboard, tInstances.kDashboard_SmartDashboard);
+              m_detected = true;
+            } else if (t.connInfo.remote_id.startsWith("shuffleboard")) {
+              HAL.report(tResourceType.kResourceType_Dashboard, tInstances.kDashboard_Shuffleboard);
+              m_detected = true;
+            } else if (t.connInfo.remote_id.startsWith("elastic")) {
+              HAL.report(tResourceType.kResourceType_Dashboard, tInstances.kDashboard_Elastic);
+              m_detected = true;
+            } else if (t.connInfo.remote_id.startsWith("Dashboard")) {
+              HAL.report(tResourceType.kResourceType_Dashboard, tInstances.kDashboard_LabVIEW);
+              m_detected = true;
+            } else if (t.connInfo.remote_id.startsWith("AdvantageScope")) {
+              HAL.report(
+                  tResourceType.kResourceType_Dashboard, tInstances.kDashboard_AdvantageScope);
+              m_detected = true;
+            } else if (t.connInfo.remote_id.startsWith("QFRCDashboard")) {
+              HAL.report(
+                  tResourceType.kResourceType_Dashboard, tInstances.kDashboard_QFRCDashboard);
+              m_detected = true;
+            } else if (t.connInfo.remote_id.startsWith("FRC Web Components")) {
+              HAL.report(
+                  tResourceType.kResourceType_Dashboard, tInstances.kDashboard_FRCWebComponents);
+              m_detected = true;
+            } else {
+              // Only report unknown if there wasn't another dashboard already reported
+              // (unknown could also be another device)
+              if (!m_detected) {
```

Comment:
This would likely report a coprocessor as an unknown dashboard before a DS laptop connects, right? I presume we're okay with that, just calling it out.

Reply:
Yes, but if a dashboard connected later, it would override the unknown. I just didn't want the unknown to override something known.